packages/components/ng-ovh-sso-auth/src/provider.js

```diff
@@ -314,6 +314,24 @@ export default function() {
         : $cookies.USERID;
     };
 
+    /**
+     * @ngdoc function
+     * @name login
+     * @methodOf ovh-angular-sso-auth.ssoAuthentication
+     *
+     * @description
+     * Set logged in data if login is done in the app
+     */
+    this.setLoggedIn = function setLoggedIn(userData) {
+      this.user = userData;
+      isLogged = true;
+      if (userData.state === NIC_STATE_ENUM.incomplete && !allowIncompleteNic) {
+        this.goToSignUpPage();
+      }
```

Comment:
**suggestions:** Seems to duplicate logic from `this.login` method just below (L343).
Maybe we can apply some refactoring.